reactioncommerce/reaction
# Add a GraphQL schema linter, fix found issues

Impact: **minor**  
Type: **feature**

## Issue
People often forget or don't know about our conventions for the GraphQL schema files, such as always including field descriptions, alphabetizing fields, and more.

## Solution
I've added [graphql-schema-linter](https://github.com/cjoudrey/graphql-schema-linter) as a dev dependency, configured it, and defined a few additional custom rules for it.

It found 300+ style issues with the `.graphql` files, which I've also fixed in this PR. Mostly missing descriptions and fields not alphabetized.

I also added this linting as a CI task.

## Breaking changes
None

## Testing
1. Verify you can run the GraphQL linter and there are no issues found: `npm run lint:gql`
2. Remove a description from a field, re-run the linter, and verify that is complains about the missing description.
3. Make sure the API still starts up OK and you can run some GraphQL queries.
